.github/workflows/cppcheck.yml

```diff
@@ -0,0 +1,76 @@
+
+name: cppcheck
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  cppcheck:
+    runs-on: ubuntu-latest
```

Comment:
ubuntu-latest is only 22.04 at this point, because 24.04 is not yet flagged stable.

Please try ubuntu-24.04 instead, in hope to get a more recent cppcheck.

Reply:
Definitely! It did not make a difference in terms of issues flagged but the versions of cppcheck were different. 

In ubuntu-latest it installed v2.7 whereas with ubuntu-24.04 we get 2.13, which is what I was already testing with locally.